File: codeforces/1555/1555C.cpp
//
//  1555C.cpp
//  Problems
//
//  Created by NicolasCardozo on 11/9/21.
//  Copyright © 2021 NicolasCardozo. All rights reserved.
//


#include <bits/stdc++.h>

using namespace std;

const int INF = 2e9 + 10;

int main() {
    int t;
    scanf("%d", &t);
    while(t--){
        int n;
        scanf("%d", &n);
        vector<vector<int>> a(2, vector<int>(n));
        for(int i=0; i<2; i++)
            for(int j=0; j<n; j++)
                scanf("%d", &a[i][j]);
        int ans = INF;
        int sum1 = 0, sum2 = 0;
        for(int i=0; i<n; i++) sum1 += a[0][i];
        for(int i=0; i<n; i++){
            sum1 -= a[0][i];
            ans = min(ans, max(sum1, sum2));
            sum2 += a[1][i];
        }
        printf("%d\n", ans);
    }
}

File: codeforces/1555/1555C.py
nTests = int(input())
for _ in range(nTests):
    m = int(input())
    matrix = [[0 for i in range(m)] for j in range(2)]
    for i in range(2):
        matrix[i] = list(map(int, input().split()))
    ans = float('inf')
    sum1, sum2 = 0, 0
    for i in range(m):
        sum1 += matrix[0][i]
    for i in range(m):
        sum1 -= matrix[0][i]
        ans = min(ans, max(sum1, sum2))
        sum2 += matrix[1][i]
    print(ans)